src/components/DonationCardModal/DonationCardModal.tsx

```diff
@@ -0,0 +1,86 @@
+import { useState } from 'react';
+import styles from './DonationCardModal.module.css';
+
+interface DonationCardModalProps {
+   title: string;
+   target: number;
+   amount: number;
+   onUpdateDonationAmount: (amount: number) => void;
+   onCloseModal: () => void;
+}
+
+function DonationCardModal({
+   title,
+   target,
+   amount,
+   onUpdateDonationAmount,
+   onCloseModal,
+}: DonationCardModalProps) {
+   const [donationAmount, setDonationAmount] = useState(0);
+
+   const handleBackgroundClick = (event: React.MouseEvent<HTMLDivElement>) => {
+      if (event.target === event.currentTarget) {
+         onCloseModal();
+      }
+   };
+
+   const handleAmountChange = (event: React.ChangeEvent<HTMLInputElement>) => {
+      setDonationAmount(Number(event.target.value));
+   };
+
+   const handleFormSubmit = (event: React.FormEvent<HTMLFormElement>) => {
+      event.preventDefault();
+      onUpdateDonationAmount(donationAmount);
+      onCloseModal();
+   };
+
+   return (
+      <div className={styles.modal} onClick={handleBackgroundClick}>
+         <form className={styles.modalContent} onSubmit={handleFormSubmit}>
+            <header className={`${styles.modalBoundary} ${styles.modalHeader}`}>
+               <div className={styles.modalHeaderContent}>
+                  <h2>{title}</h2>
+                  <button
+                     type="button"
+                     className={styles.xButton}
+                     onClick={onCloseModal}>
+                     &times;
+                  </button>
+               </div>
+            </header>
+            <main className={styles.sectionContent}>
+               <div className={styles.formGroup}>
+                  <label htmlFor="amount">Select Donation Amount</label>
+                  <input
+                     id="amount"
+                     type="range"
+                     min="0"
+                     max={target - amount}
+                     step="50"
+                     aria-required="true"
+                     onChange={handleAmountChange}
+                  />
```

Comment:
This has the logic for the "amount" part as one of the solutions that we discussed about :)